### ISSUE TITLE
CRIT uncaptured python exception

### ISSUE THREAD

**Issue by Tatikoma** [Supervisor/supervisor#853]
I got error on my production server (cannot reproduce):
```
2016-11-17 09:14:15,546 CRIT uncaptured python exception, closing channel <POutputDispatcher at 139665565490424 for <Subprocess at 139665565913240 with name router0 in state RUNNING> (stdout)> (<type 'exceptions.OSError'>:[Errno 2] No such file or directory [/usr/lib/python2.7/dist-packages/supervisor/supervisord.py|runforever|233] [/usr/lib/python2.7/dist-packages/supervisor/dispatchers.py|handle_read_event|231] [/usr/lib/python2.7/dist-packages/supervisor/dispatchers.py|record_output|187] [/usr/lib/python2.7/dist-packages/supervisor/dispatchers.py|_log|141] [/usr/lib/python2.7/dist-packages/supervisor/loggers.py|info|273] [/usr/lib/python2.7/dist-packages/supervisor/loggers.py|log|291] [/usr/lib/python2.7/dist-packages/supervisor/loggers.py|emit|186] [/usr/lib/python2.7/dist-packages/supervisor/loggers.py|doRollover|220])
```

After this error process "router0" looked okay in supervisor, but in fact the process did not work. Also the process was no corresponding log file (logging is configured in supervisor). Everything was working correctly, after restarting router0 process.

Config file for router0 process is simple:
```
[program:router]
command=php /home/user/router.php
directory=/home/user
user=tatikoma
process_name=%(program_name)s%(process_num)s
numprocs=1
autostart=true
autorestart=true
startsecs=5
startretries=3
stopsignal=TERM
stopwaitsecs=10
umask=0002
redirect_stderr=true
stdout_logfile=/tmp/%(program_name)s%(process_num)s.log
stdout_logfile_maxbytes=1MB
stdout_logfile_backups=10
stdout_capture_maxbytes=1MB
```

```
# dpkg -l | grep supervisor
ii  supervisor                     3.0r1-1                           all          A system for controlling process state
# uname -a
Linux smpp.bringo.lan 3.16.0-4-amd64 #1 SMP Debian 3.16.36-1+deb8u2 (2016-10-19) x86_64 GNU/Linux
```

**Comment by mnaberez** [comment]
Expanding the log message above:

```
2016-11-17 09:14:15,546 CRIT uncaptured python exception, closing channel <POutputDispatcher at 139665565490424 for <Subprocess at 139665565913240 with name router0 in state RUNNING> (stdout)> 
(<type 'exceptions.OSError'>:[Errno 2] No such file or directory 
[/usr/lib/python2.7/dist-packages/supervisor/supervisord.py|runforever|233] 
[/usr/lib/python2.7/dist-packages/supervisor/dispatchers.py|handle_read_event|231] 
[/usr/lib/python2.7/dist-packages/supervisor/dispatchers.py|record_output|187] 
[/usr/lib/python2.7/dist-packages/supervisor/dispatchers.py|_log|141] 
[/usr/lib/python2.7/dist-packages/supervisor/loggers.py|info|273] 
[/usr/lib/python2.7/dist-packages/supervisor/loggers.py|log|291] 
[/usr/lib/python2.7/dist-packages/supervisor/loggers.py|emit|186] 
[/usr/lib/python2.7/dist-packages/supervisor/loggers.py|doRollover|220])
```

**Comment by mnaberez** [comment]
Fixed in 16b46ac019c1b9f5a1aee030fcc3b6f2bef33f42 (#402) and released in Supervisor 3.1.0 (2014-07-29).
